pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -322,22 +337,92 @@ func (c *podIPMetricsCollector) setupDirectoryWatcher(dir string) error {
 	return nil
 }
 
+func (c *podIPMetricsCollector) countTotalIPs(subnet *net.IPNet) (uint64, error) {
+	ones, bits := subnet.Mask.Size()
+	if bits == 0 {
+		return 0, fmt.Errorf("invalid IP subnet mask %v", subnet.Mask)
+	}
+	if ones == bits {
+		ones = bits - 1
+	}
+	return uint64(1 << (bits - ones)), nil
+}
+
+func (c *podIPMetricsCollector) calculateAssignedIP(subnetFile string) error {
+	var assignedIPv4, assignedIPv6 uint64
+	networkConfigList, err := libcni.ConfListFromFile(subnetFile)
+	if err != nil {
+		glog.Errorf("Error reading network config from file %s: %v", subnetFile, err)
+		return err
+	}
+
+	var ipamConfig *allocator.IPAMConfig
+	for _, plugin := range networkConfigList.Plugins {
+		if plugin.Network.Type == "gke" {
+			ipamConfig, _, err = allocator.LoadIPAMConfig(plugin.Bytes, "")
+			if err != nil {
+				glog.Errorf("Error loading IPAM config: %v", err)
+				return err
+			}
+			break
+		}
+	}
+
+	for _, ranges := range ipamConfig.Ranges {
+		for _, subnets := range ranges {
+			convertedSubnet := &net.IPNet{
```

Comment:
Do we have to convert this or can we just stick to `github.com/containernetworking/cni/pkg/types.IPNet`? Also consider the `rangeStart` and `rangeEnd` comment above.

Reply:
Sure. Let me also import the package `types` as well.

Reply:
Done